src/main/scala/com/fulcrumgenomics/umi/GroupReadsByUmi.scala

```diff
@@ -83,23 +83,23 @@ object GroupReadsByUmi {
         tmp
       }
       else {
-        val lib     = library(rec)
-        val chrom   = rec.refIndex
-        val recNeg  = rec.negativeStrand
-        val recPos  = if (recNeg) rec.unclippedEnd else rec.unclippedStart
+        val lib       = library(rec)
+        val chrom     = rec.refIndex
+        val mateChrom = rec.mateRefIndex
+        val recNeg    = rec.negativeStrand
+        val recPos    = if (recNeg) rec.unclippedEnd else rec.unclippedStart
 
         val (mateNeg, matePos) = if (!rec.paired) (false, Int.MaxValue) else {
-          require(rec.refIndex == rec.mateRefIndex, s"Mate on different chrom for ${rec.name}.")
           val neg = rec.mateNegativeStrand
           val pos = if (neg) SAMUtils.getMateUnclippedEnd(rec.asSam) else SAMUtils.getMateUnclippedStart(rec.asSam)
           (neg, pos)
         }
 
-        val result = if (recPos < matePos || (recPos == matePos && !recNeg)) {
+        val result = if (chrom < mateChrom || chrom == mateChrom && (recPos < matePos || (recPos == matePos && !recNeg))) {
```

Comment:
Just adding one extra bracket to make 100% clear the precedence of `||` and `&&`:
```suggestion
        val result = if (chrom < mateChrom || (chrom == mateChrom && (recPos < matePos || (recPos == matePos && !recNeg)))) {
```